robottelo/ui/contentviews.py

```diff
@@ -1,126 +1,86 @@
-            self.wait_for_ajax()
-            self.find_element(locators["contentviews.search_button"]).click()
-            self.wait_for_ajax()
-            strategy, value = locators["contentviews.select_filter_name"]
-            element = self.wait_until_element((strategy,
-                                               value % filter_name))
-            if element:
-                element.click()
-                self.wait_for_ajax()
-            else:
-                raise UIError(
-                    'Could not find filter with name "{0}"'.format(filter_name)
-                )
-        else:
+        if not element:
             raise UIError(
                 'Could not find the selected CV "{0}"'.format(cv_name)
             )
 
+        element.click()
+        self.wait_for_ajax()
+        self.click(tab_locators['contentviews.tab_content'])
+        self.click(locators['contentviews.content_filters'])
+        self.text_field_update(
+            locators['contentviews.search_filters'], filter_name)
+        self.wait_for_ajax()
+        self.click(locators['contentviews.search_button'])
+        strategy, value = locators['contentviews.select_filter_name']
+        self.click((strategy, value % filter_name))
+
     def create(self, name, label=None, description=None, is_composite=False):
         """Creates a content view"""
-        self.wait_until_element(locators["contentviews.new"]).click()
+        self.click(locators['contentviews.new'])
 
-        if self.wait_until_element(common_locators["name"]):
+        if self.wait_until_element(common_locators['name']):
             self.find_element(
-                common_locators["name"]).send_keys(name)
+                common_locators['name']).send_keys(name)
             timeout = 60 if len(name) > 50 else 30
             self.wait_for_ajax(timeout)
 
             if label is not None:
-                self.find_element(common_locators["label"]).send_keys(label)
-
+                self.find_element(common_locators['label']).send_keys(label)
             if description is not None:
                 self.find_element(
-                    common_locators["description"]).send_keys(description)
-
+                    common_locators['description']).send_keys(description)
             if is_composite:
-                self.find_element(
-                    locators["contentviews.composite"]).click()
-
-            self.wait_for_ajax()
-            self.wait_until_element(common_locators["create"]).click()
+                self.click(locators['contentviews.composite'])
             self.wait_for_ajax()
+            self.click(common_locators['create'])
         else:
             raise UIError(
                 'Could not create new content view "{0}"'.format(name)
             )
 
-    def delete(self, name, really):
+    def delete(self, name, really=True):
         """Deletes an existing content view."""
         element = self.search(name)
-
         if element:
```

Comment:
Hm... The logic of this function looks suspicious for me. As there's no `else` statement it looks like operation will be successful even if CV wasn't actually found and wasn't deleted.
I briefly ran through similar functions in other ui modules, and most of them are using `self.delete_entity()` instead of this approach (there're few similar implementations though).
Maybe it's time to refactor this `delete()`?